astroquery/desi/core.py

```diff
@@ -0,0 +1,97 @@
+
+    def query_region(self, coordinates, radius, *, data_release=9):
+        """
+        Queries a region around the specified coordinates.
+
+        Parameters
+        ----------
+        coordinates : `astropy.coordinates`
+            coordinates around which to query.
+        radius : `astropy.units.Quantity`
+            the radius of the cone search.
+        data_release: int
+            the data release of the LegacySurvey to use.
+
+        Returns
+        -------
+        response : `astropy.table.Table`
+        """
+
+        tap_service = vo.dal.TAPService(conf.tap_service_url)
+        coordinates_transformed = coordinates.transform_to(coord.ICRS)
+
+        qstr = (f"SELECT all * FROM ls_dr{data_release}.tractor WHERE "
+               f"dec>{coordinates_transformed.dec.deg - radius.deg} and "
+               f"dec<{coordinates_transformed.dec.deg + radius.deg} and "
+               f"ra>{coordinates_transformed.ra.deg - radius.deg / np.cos(coordinates_transformed.dec.deg * np.pi / 180.)} and "
+               f"ra<{coordinates_transformed.ra.deg + radius.deg / np.cos(coordinates_transformed.dec.deg * np.pi / 180)}")
+
+        tap_result = tap_service.run_sync(qstr)
+        tap_result = tap_result.to_table()
+        # filter out duplicated lines from the table
+        mask = tap_result['type'] != 'D'
+        filtered_table = tap_result[mask]
+
+        return filtered_table
+
+    def get_images(self, position, pixels, radius, *, data_release=9, show_progress=True, image_band='g'):
+        """
+        Downloads the images for a certain region of interest.
+
+        Parameters
+        -------
+        position: `astropy.coordinates`.
+            coordinates around which we define our region of interest.
+        radius: `astropy.units.Quantity`.
+            the radius of the cone search.
+        data_release: int
+            the data release of the LegacySurvey to use.
+
+        Returns
+        -------
+        list: A list of `astropy.io.fits.HDUList` objects.
+        """
+
+        position_transformed = position.transform_to(coord.ICRS)
+
+        image_size_arcsec = radius.arcsec
+        pixsize = 2 * image_size_arcsec / pixels
+
+        image_url = (f"{conf.legacysurvey_service_url}?"
+                    f"ra={position_transformed.ra.deg}&"
+                    f"dec={position_transformed.dec.deg}&"
+                    f"size={pixels}&"
+                    f"layer=ls-dr{data_release}&"
+                    f"pixscale={pixsize}&"
+                    f"bands={image_band}")
+
+        file_container = commons.FileContainer(image_url, encoding='binary', show_progress=show_progress)
+
+        try:
+            fits_file = file_container.get_fits()
+        except (requests.exceptions.HTTPError, urllib.error.HTTPError) as e:
+            fits_file = None
+            warnings.warn(f"{str(e)} - Problem retrieving the file at the url: {image_url}", NoResultsWarning)
```

Comment:
are you sure you don't want to raise this as an exception? Why and when does this return an HTTPError?

Reply:
An HTTPError happens in the case of not valid inputs, more specifically when these are outside of the survey coverage.

We, for our internal applications, want to display a warning message in case this situation occurs.

Reply:
yes, my point is that if the URL throws an error then I think we should throw it, too. E.g. the user expects a file as a return, but here they get a valid return yet no files.
Is there any use case where this is more useful for the users than to receive an error?

Reply:
The server returns an `HTTPError` in the case of  internal error of the legacysurvey but also in the case I mentioned (ie case of not valid inputs, outside of the survey coverage), so at the moment is impossible for us to distinguish these two situations.

For this reason, we decided just to show the message in the warning to the user in both of those cases.

Reply:
Hi @bsipocz , what do you think? Does it make sense to you?

Reply:
It's OK to customize the message but my point still stands, it should the raised. I don't see a use case where a warning rather than an error is preferred after all the query errored out and the `None` return is not really a valid results for these cases (`None` return would be fine if there is a valid query and there is no match for ti in the DB, etc.)